core/src/test/scala/unit/kafka/utils/JaasTestUtils.scala

```diff
@@ -122,6 +136,19 @@ object JaasTestUtils {
   val KafkaScramAdmin = "scram-admin"
   val KafkaScramAdminPassword = "scram-admin-secret"
 
+  val isIBMJdk = System.getProperty("java.vendor").contains("IBM")
+  val serviceName = "kafka"
+
+  def saslConfigs(saslProperties: Option[Properties]): Properties = {
+    val result = saslProperties match {
+      case Some(properties) => properties
+      case None => new Properties
+    }
+    if (isIBMJdk)
+      result.put(KafkaConfig.SaslKerberosServiceNameProp, JaasTestUtils.serviceName)
```

Comment:
Ideally we would take a JaasSection as a parameter and retrieve the service name from there, but not sure how easy that is. If complicated, then this approach may work, but we should at least check if the config is already present before setting it. And we should have a comment explaining why this is needed.